src/encoded/static/components/motifs.js

```diff
@@ -142,13 +188,39 @@ export const Motifs = (props) => {
 
     // Filter results to find ones with associated PWM data
     const pwmLinkListFull = results.filter(d => d.documents && d.documents[0] && d.documents[0].document_type === 'position weight matrix');
-    let pwmLinkList = [];
-    if (limit > 0 && pwmLinkListFull.length !== 0) {
-        pwmLinkList = pwmLinkListFull.slice(0, limit);
+
+    const groupedList = _.groupBy(pwmLinkListFull, link => `${link.documents[0]['@id']}#${link.targets.join('-')}`);
+
+    let groupedListMapped = _.map(groupedList, group => ({
+        pwm: group[0].documents[0]['@id'],
+        href: group[0].documents[0].attachment.href,
+        targets: group[0].targets.join(', '),
+        accessions: _.pluck(group, 'file'),
+        datasets: _.pluck(group, 'dataset'),
+        start: group[0].start,
+        end: group[0].end,
+        strand: group[0].strand,
+        biosamples: _.pluck(group, 'biosample_ontology').map(d => d.term_name),
+        description: group[0].description ? group[0].description : null,
+    }));
+
+    groupedListMapped = _.sortBy(groupedListMapped, o => o.targets);
+
```

Comment:
I do feel a little skeptical that, in a few months, even _you’ll_ remember what this is for and how it works. Seems pretty complex but not documented.

Reply:
I added a bunch of comments, hopefully this is clearer now.